components/casousel/Carousel.tsx

```diff
@@ -0,0 +1,63 @@
+"use client"
+import { useState, useEffect } from 'react';
+
+const Carousel = () => {
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const slides = [
+    {
+      img: 'https://images.unsplash.com/photo-1542291026-7eec264c27ff?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwzNjUyOXwwfDF8c2VhcmNofDF8fHByb2R1Y3R8ZW58MHx8fHwxNjI2MTg5OTUx&ixlib=rb-1.2.1&q=80&w=400',
+      title: 'Premium Nike Footwear',
+      price: '$100',
+      rating: '★★★★★',
+      link: '#',
+    },
+    {
+      img: 'https://images.unsplash.com/photo-1526170375885-4d8ecf77b99f?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwzNjUyOXwwfDF8c2VhcmNofDJ8fHByb2R1Y3R8ZW58MHx8fHwxNjI2MTg5OTUx&ixlib=rb-1.2.1&q=80&w=400',
+      title: 'Vintage Film Camera',
+      price: '$200',
+      rating: '★★★★★',
+      link: '#',
+    },
+    {
+      img: 'https://images.unsplash.com/photo-1502672260266-1c1ef2d93688?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwzNjUyOXwwfDF8c2VhcmNofDEwfHByb2R1Y3R8ZW58MHx8fHwxNjI2MTg5OTUx&ixlib=rb-1.2.1&q=80&w=400',
+      title: 'Cozy Room Essentials',
+      price: '$300',
+      rating: '★★★☆☆',
+      link: '#',
+    },
+  ];
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setCurrentIndex((prevIndex) =>
+        prevIndex === slides.length - 1 ? 0 : prevIndex + 1
+      );
+    }, 3000); // Change slide every 3 seconds
+
+    return () => clearInterval(interval);
+  }, [slides]); // Added slides as a dependency to useEffect
+
```

Comment:
- That's not the way correct way of using shadcn components, never edit shadcn components itself
- Please consider checking shadcn docs about how to implement carousel